internal/preflight/dryrun.go

```diff
@@ -12,25 +15,63 @@ type CreationDryRun struct {
 }
 
 func NewCreationDryRun(client client.Writer) *CreationDryRun {
-	return &CreationDryRun{
-		client: client,
-	}
+	return &CreationDryRun{client: client}
 }
 
-func (p *CreationDryRun) Check(
-	ctx context.Context, _, obj client.Object,
-) (violations []Violation, err error) {
+func (p *CreationDryRun) Check(ctx context.Context, _, obj client.Object) (violations []Violation, err error) {
 	defer addPositionToViolations(ctx, obj, &violations)
 
-	drerr := p.client.Create(ctx, obj.DeepCopyObject().(client.Object), client.DryRunAll)
-	if errors.IsAlreadyExists(drerr) {
+	err = p.client.Create(ctx, obj.DeepCopyObject().(client.Object), client.DryRunAll)
+
+	if k8serrs.IsAlreadyExists(err) {
+		obj := obj.DeepCopyObject().(*unstructured.Unstructured)
+		err = p.client.Patch(ctx, obj, client.MergeFrom(obj), client.DryRunAll)
+	}
```

Comment:
Can we please turn this around please, checking Patch by default and only testing Create if not found?
I think PKO will do way more patching than creation, which would make this more efficient.